src/GraphQL.Client.LocalExecution/GraphQLLocalExecutionClient.cs

```diff
@@ -95,16 +93,15 @@ private async Task<ExecutionResult> ExecuteAsync(GraphQLRequest request, Cancell
 
             var deserializedRequest = JsonConvert.DeserializeObject<GraphQLRequest>(serializedRequest);
             var inputs = deserializedRequest.Variables != null
-                ? (JObject.FromObject(request.Variables, JsonSerializer.Create(_variablesSerializerSettings)) as JObject)
-                .ToInputs()
+                ? _documentSerializer.ReadNode<Inputs>(JObject.FromObject(request.Variables, JsonSerializer.Create(_variablesSerializerSettings)))
                 : null;
```

Comment:
```suggestion
            var deserializedRequest = _documentSerializer.Deserialize<Transport.GraphQLRequest>(serializedRequest);
```

Reply:
It seems that GraphQLRequest (as defined in the GraphQL Client repo) is a dictionary, where the variables are a native C# object.  So it's serialized to a string and deserialized again.  But we can take a shortcut and deserialize it via the GraphQL.NewtonsoftJson deserializer directly to the new GraphQL.Transport.GraphQLRequest class, which has already read the variables as an Inputs object.

Reply:
done